tt_eager/tt_dnn/op_library/transformer_tms/transformer_tms.hpp

```diff
@@ -216,6 +224,44 @@ inline Tensor ssm_1d_sum_reduce(const Tensor &input_tensor_a, const MemoryConfig
     return output_tensors.at(0);
 }
 
+struct SSMPrefixScan {
+    MemoryConfig output_mem_config;
+    DataType output_dtype;
+    MathFidelity math_fidelity;
+
+    const uint32_t HIDDEN_SIZE = 5120;
+    const uint32_t LATENT_SIZE = 32;
+
+    void validate(const std::vector<Tensor>& input_tensors) const;
+    std::vector<Shape> compute_output_shapes(const std::vector<Tensor>& input_tensors) const;
+    std::vector<Tensor> create_output_tensors(const std::vector<Tensor>& input_tensors) const;
+    operation::ProgramWithCallbacks create_program(
+        const std::vector<Tensor>& input_tensors, std::vector<Tensor>& output_tensors) const;
+    tt::stl::reflection::Attributes attributes() const;
+};
+
+inline Tensor ssm_prefix_scan(
+    const Tensor& a,
+    const Tensor& bx,
+    const MemoryConfig& mem_config,
+    std::optional<const DataType> output_dtype = std::nullopt,
+    MathFidelity math_fidelity = MathFidelity::HiFi4) {
+    std::vector<Tensor> output_tensors = {Tensor(operation::get_workers_for_op_output({a, bx}))};
+    operation::launch_op(
+        [mem_config, output_dtype, math_fidelity](
+            const std::vector<Tensor>& input_tensors,
+            const std::vector<std::optional<const Tensor>>& optional_input_tensors,
+            const std::vector<std::optional<Tensor>>& optional_output_tensors) mutable -> std::vector<Tensor> {
+            const auto& a = input_tensors.at(0);
+            const auto& bx = input_tensors.at(1);
+            return operation::run(
+                SSMPrefixScan{mem_config, output_dtype.value_or(a.get_dtype()), math_fidelity}, input_tensors);
```

Comment:
Pass in `HIDDEN_SIZE` and `LATENT_SIZE`

Reply:
See my other comment - I wasn't actually using these anywhere so I removed them.